module/applications/compendium-browser.mjs

```diff
@@ -0,0 +1,416 @@
+      }
+    }
+  };
+
+  /* -------------------------------------------- */
+
+  /** @override */
+  static PARTS = {
+    documentClass: {
+      id: "sidebar-class",
+      classes: ["sidebar-part"],
+      template: "systems/dnd5e/templates/compendium/browser-sidebar-class.hbs"
+    },
+    types: {
+      id: "sidebar-types",
+      classes: ["sidebar-part"],
+      template: "systems/dnd5e/templates/compendium/browser-sidebar-types.hbs"
+    },
+    results: {
+      id: "results",
+      classes: ["results"],
+      template: "systems/dnd5e/templates/compendium/browser-results.hbs",
+      templates: ["systems/dnd5e/templates/compendium/browser-entry.hbs"],
+      scrollable: [""]
+    },
+    footer: {
+      id: "footer",
+      classes: ["footer"],
+      template: "systems/dnd5e/templates/compendium/browser-footer.hbs"
+    }
+  };
+
+  /* -------------------------------------------- */
+  /*  Properties                                  */
+  /* -------------------------------------------- */
+
+  /**
+   * Currently define filters.
+   */
+  #filters;
+
+  /**
+   * Current filters selected.
+   * @type {CompendiumBrowserFilters}
+   */
+  get currentFilters() {
+    const filters = foundry.utils.mergeObject(
+      this.#filters,
+      this.options.filters.locked,
+      { inplace: false }
+    );
+    filters.documentClass ??= "Item";
+    return filters;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Fetched results.
+   * @type {Promise<object[]|Document[]>}
+   */
+  #results;
+
+  /* -------------------------------------------- */
+  /*  Rendering                                   */
+  /* -------------------------------------------- */
+
+  /** @override */
+  _onFirstRender(context, options) {
+    const sidebar = document.createElement("form");
+    sidebar.classList.add("sidebar", "flexcol");
+    sidebar.replaceChildren(...this.element.querySelectorAll(".sidebar-part"));
+    this.element.querySelector(".window-content").insertAdjacentElement("afterbegin", sidebar);
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async _preparePartContext(partId, context, options) {
+    await super._preparePartContext(partId, context, options);
+    switch (partId) {
+      case "documentClass":
+      case "types": return this._prepareSidebarContext(partId, context, options);
+      case "results": return this._prepareResultsContext(context, options);
+    }
+    return context;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Prepare the sidebar context.
+   * @param {string} partId                        The part being rendered.
+   * @param {ApplicationRenderContext} context     Shared context provided by _prepareContext.
+   * @param {HandlebarsRenderOptions} options      Options which configure application rendering behavior.
+   * @returns {Promise<ApplicationRenderContext>}  Context data for a specific part.
```

Comment:
I'm guessing this is intended to be `@protected` since it's underscore-prefixed. The same for the methods below this one.